shm/Player.hpp

```diff
@@ -8,22 +8,22 @@
 
 class Player : public Delegate {
 private:
-    std::unique_ptr<Ship> ship_;
+    std::shared_ptr<Ship> ship_;
```

Comment:
And why was this changed from unique_ptr?

Reply:
it was causing error that object does not exist anymore, im trying to figure it out now